feat(#5579): add topics to reports

elasticspoon opened the pull request:
Fixes https://github.com/rubyforgood/casa/issues/5579
Fixes #3028
Depends on
- [x] #5590
- [x] #5593

## Overview
In #5388 I added` ContactTopics` and `ContactTopicAnswers` to `CaseContacts`. This allowed `CasaOrg`s to set topics and Users to create `ContactTopicAnswers` on specific `CaseContacts`. 

Now we want to have those answers appear in the court reports.

This PR modifies `CourtReportContext` to include topic information, as well as modifies `CourtReport` templates to accept that information.

## Process

Topics are taken out of the database and massaged into the following shape:

```rb
  [
    {topic: "Question 1", details: "Details 1", answers: [
      {date: "12/02/20", medium: "Type A1, Type B1", value: "Answer 1"},
      {date: "12/03/20", medium: "Type A2, Type B2", value: "Answer 3"}
    ]},
    {topic: "Question 2", details: "Details 2", answers: [
      {date: "12/02/20", medium: "Type A1, Type B1", value: "Answer 2"},
      {date: "12/04/20", medium: "Type A3, Type B3", value: "Answer 5"}
    ]},
    {topic: "Question 3", details: "Details 3", answers: [
      {date: "12/03/20", medium: "Type A2, Type B2", value: "No Answer Provided"},
      {date: "12/04/20", medium: "Type A3, Type B3", value: "No Answer Provided"}
    ]}
  ]

```

That info gets filled in via document templates like so

![image](https://github.com/rubyforgood/casa/assets/14540596/f0cf495f-114c-4788-b0a1-4c9fe1a00d27)

Just to label the parts:
![image](https://github.com/rubyforgood/casa/assets/14540596/0d251de2-6a65-4a5f-906d-087c7d969c67)

1. Contact Topic
2. Topic details
3. Medium of contact
4. Contact Date
5. Answer to topic on that contact

elasticspoon committed:
feat: add contact topics to reports

In #5388 I added ContacTopics and ContactTopicAnswers to CaseContacts.
This allowed CasaOrgs to set topics and Users to create
ContactTopicAnswers on specific CaseContacts.

Now we want to have those answers appear in the court reports.

This PR modifies CourtReportContext to include topic information, as
well as modifies CourtReport templates to accept that information.

Modified the templates to render the correct information. If an org has
no topics the fallback is the old values.

Additionally if a org has no topics the form/details page should not
show the card for court topics.

Adds after_party update task to update prod organizations with new
templates.

Commit:
Fix table issue for Howard

Commit:
Howard template fix

elasticspoon committed:
fix: ci failure

unclear what caused it.

elasticspoon committed:
Merge branch 'main' into add-topics-to-reports

compwron commented:
Fix merge conflict
```
This branch has conflicts that must be resolved
Use the [web editor](https://github.com/rubyforgood/casa/pull/5601/conflicts) or the to resolve conflicts.
Conflicting files
app/views/case_contacts/form/expenses.html.erb
```

compwron commented:
Nice labeled diagram :)

compwron commented on app/documents/templates/prince_george_report_template.docx:
Why did this one get smaller? Seems strange and maybe not good
<img width="820" alt="Screenshot 2024-05-14 at 8 50 24 AM" src="https://github.com/rubyforgood/casa/assets/578159/df26cf19-e5fd-48d4-9c7a-28e91c7e0102">

elasticspoon replied:
No clue. The template got longer so that's weird.
old: ![image](https://github.com/rubyforgood/casa/assets/14540596/eb1fe598-1904-4a3f-aa91-c3e18ff8f95d)

new: ![image](https://github.com/rubyforgood/casa/assets/14540596/a1692e7d-3af4-43d8-aaf7-52b8555d605e)

compwron commented on app/models/case_court_report_context.rb:
```diff
@@ -88,6 +89,51 @@ def org_address(path_to_template)
     @volunteer.casa_org.address if @volunteer && is_default_template
   end
 
+  # Sample output
```
good comment :)

compwron commented on app/models/case_court_report_context.rb:
```diff
@@ -88,6 +89,51 @@ def org_address(path_to_template)
+  # ]}
+  # }
+  def court_topics
+    topics = ContactTopic
```
nice query. It could be slow but we are working with small numbers I think we're fine

compwron reviewed (approved):
good tests :)

elasticspoon committed:
fix: dependabot failures due to no secrets

Dependabot does not get passed secrets. Just don't run the
actions that require secrets if they aren't passed in.

elasticspoon committed:
fixup!

elasticspoon committed:
Merge branch 'main' into add-topics-to-reports

elasticspoon committed:
Merge branch 'fix-dependabot-ci' into add-topics-to-reports

compwron reviewed (approved):
Heck yes
